### ISSUE TITLE
Add support for each-expressions

### ISSUE THREAD

**Issue by asos-neildownes** [sharpliner/sharpliner#281]
Would it be possible to add the ability to generate "each-expressions" in the auto generated yaml files?  

I'd like to be able to iterate over a collection, like an array of environments, and use each environment variable as an input parameter to a yaml template.  Please see example below.

```
- ${{ each env in parameters.environments}}:
  - template: ../stages/provision.yml
    parameters:
      environment: ${{env}}
      ${{ if eq(env.deploymentEnvironmentName, '') }}:
        deploymentEnvironment: ${{ parameters.applicationName }} - ${{ env.name }}
      ${{else}}:
        deploymentEnvironment: ${{ env.deploymentEnvironmentName }}
      regions: ${{parameters.regions}}
      pipelinePath: ${{parameters.pipelinePath}}
      applicationName: ${{parameters.applicationName}}
```

**Comment by premun** [comment]
Thanks for opening this issue, this is a missing feature. This should be easy to add, I will have a look at it next week.

**Comment by asos-neildownes** [comment]
Thanks much appreciated

**Comment by premun** [comment]
Alright, I made some progress and made your example work:
https://github.com/sharpliner/sharpliner/blob/55b9fbb12e9610df0013a79792a195982d47f94a/tests/Sharpliner.Tests/AzureDevOps/ConditionedExpressions/EachExpressionTests.cs

I only need to verify everything and then I will push this change and make a new version of the library.

**Comment by premun** [comment]
Apologies, I fell sick. But the problem seems a bit more complex than I thought. It's about how ifs and each blocks can be mixed, it makes it quite hard to have both.

**Comment by asos-neildownes** [comment]
Hello, perhaps delivering the each blocks without if/else will still provide value to developers.
Thanks in advance

**Comment by premun** [comment]
Ah, I was able to find a nice solution which makes it work with everything. Will post a PR + new version shortly.

**Comment by premun** [comment]
@asos-neildownes there will be a new version `1.5.7` of the package that should support `each`.
I tried to do it fast so it's possible there might be some convoluted cases where it doesn't work. For instance, nested each blocks don't work but maybe I can add the support as a follow up and you are hopefully unblocked now.

Let me know how it works.

**Comment by premun** [comment]
I am closing this in favour of a new issue that only describes the problems with nested conditioned blocks
